src/AutoMapper/IProfileExpression.cs

```diff
@@ -53,6 +55,14 @@ public interface IProfileExpression
         IMappingExpression CreateMap(Type sourceType, Type destinationType, MemberList memberList);
 
         /// <summary>
+        /// Transforms PropertyMap with source and destination Type
+        /// </summary>
+        /// <typeparam name="TSource"></typeparam>
+        /// <typeparam name="TDestination"></typeparam>
+        /// <param name="transformExpression"></param>
+        void TransformPropertyMap<TSource>(Expression<Func<TSource, TSource>> transformExpression);

```

Comment:
Oh if you want the IfThen PR then no this can't do it I think.  What you would do is `ForAllPropertyMaps` for the condition to map it.  If you want a condition to make the modification, that needs to be another parameter passed over.  It's just a question of what you want passed over.  Want pre set rules based off source type, or want to pass something like PropertyMap, or MemberInfo over, so that they can make any condition they like?